test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java

```diff
@@ -25,50 +25,82 @@
-            and also check if the test works OK if the mouse wheel is rotated very slow.
+            the test will print mouse wheel event messages in the format
+            <b> [Event#, WheelRotation, PreciseWheelRotation]</b> into the logging
+            panel below the instruction window.<br> <br>
 
-            If the above is true press PASS, else FAIL.
+            A hi-res mouse is one which produces MouseWheelEvents having
+            <b>preciseWheelRotation &lt 1.</b> <br>
+            When preciseWheelRotation adds up to 1,wheelRotation becomes 1. <br>
+            You should see a few events where preciseWheelRotation < 1 & wheelRotation = 0
+            followed by a event where preciseWheelRotation = 1 & wheelRotation = 1.<br> <br>
+
+            Check if the test works OK when the mouse wheel is rotated very slow.<br> <br>
+            Please press PASS if above is true, else FAIL. <br> <br>
+
+            <hr>
+            PS: If you don't see events with preciseWheelRotation < 1,
+            then the mouse doesn't support high-resolution scrolling.
+            A warning is shown in the logging area if you are not using a hi-res mouse.
+            <br> <br>
+            </body>
+            </html>
             """;
 
-    public static void main (String[] args) throws Exception {
+    public static void main(String[] args) throws Exception {
         PassFailJFrame.builder()
                 .title("Test Instructions")
                 .instructions(INSTRUCTIONS)
-                .rows((int) INSTRUCTIONS.lines().count() + 2)
-                .columns(45)
-                .logArea(8)
+                .rows(20)
+                .columns(50)
+                .testTimeOut(10)
+                .logArea(10)
                 .testUI(AWTPanelSmoothWheel::createUI)
                 .build()
                 .awaitAndCheck();
     }
 
-    private static Frame createUI () {
+    private static Frame createUI() {
         Frame frame = new Frame("Test Wheel Rotation");
         Panel panel = new Panel();
         panel.setBackground(Color.GREEN);
-        panel.addMouseWheelListener(e -> PassFailJFrame.log(e.toString()));
-        frame.setSize (200, 200);
+        panel.addMouseWheelListener(e -> {
+            if (e.getScrollType() == MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+                PassFailJFrame.log("WheelEvent#"+ ++wheelEventCount
+                        + " --- Wheel Rotation: " + e.getWheelRotation()
+                        + " --- Precise Wheel Rotation: "
+                        + String.format("%.2f", e.getPreciseWheelRotation()));
+                if (e.getPreciseWheelRotation() < 1) {
+                    hiResWheelCount++;
+                }
+                if (wheelEventCount >= 5 && hiResWheelCount == 0) {
+                    PassFailJFrame.log("WARNING !!! You might not be using a high-res mouse.");
+                }
+            }
```

Comment:
This logging style is easier to parse, thanks.

You have nearly all the data to make the test semi-automatic. What's missing is `wheelRotationCount`.

[This diff](https://github.com/aivanov-jdk/jdk/commit/04c80c519d1c116663ab10a177bff1c4b6924091) implements this feature. For convenience, I'll paste the diff inline too:
```diff
diff --git a/test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java b/test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java
index 1097d1dcb04..f3dd8294e8b 100644
--- a/test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java
+++ b/test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java
@@ -39,7 +39,9 @@

 public class AWTPanelSmoothWheel {
     private static int wheelEventCount = 0;
+    private static int wheelRotationCount = 0;
     private static int hiResWheelCount = 0;
+
     private static final String INSTRUCTIONS = """
             <html>
             <body>
@@ -92,12 +94,21 @@ private static Frame createUI() {
                         + " --- Wheel Rotation: " + e.getWheelRotation()
                         + " --- Precise Wheel Rotation: "
                         + String.format("%.2f", e.getPreciseWheelRotation()));
+                if (e.getWheelRotation() >= 1) {
+                    wheelRotationCount += e.getWheelRotation();
+                }
                 if (e.getPreciseWheelRotation() < 1) {
                     hiResWheelCount++;
                 }
                 if (wheelEventCount >= 5 && hiResWheelCount == 0) {
                     PassFailJFrame.log("WARNING !!! You might not be using a high-res mouse.");
                 }
+                if (wheelRotationCount > 5
+                    && (hiResWheelCount / 2 >= wheelRotationCount)) {
+                    PassFailJFrame.log("The test passes: hiResWheelCount = " + hiResWheelCount
+                                       + " wheelRotationCount = " + wheelRotationCount);
+                    PassFailJFrame.forcePass();
+                }
             }
         });
         frame.setSize (400, 200);
```